### ISSUE TITLE
Error when trying to get definition without results

### ISSUE THREAD

**Issue by Hoffs** [Hoffs/omnisharp-extended-lsp.nvim#11]
For example when navigating into `Func<T, TResult>` trying to get definition of delegate causes an error. Should check if any locations were returned first.

**Comment by gramatus** [comment]
I think I have the same problem (might be not quite the same). E.g. at this point in my code, in VsCode  I will get a message "no definition found":
![image](https://user-images.githubusercontent.com/93591857/231216606-4648ed22-be21-4963-a854-cdf9715e02ca.png)

Now in vim I get this message:
![image](https://user-images.githubusercontent.com/93591857/231217564-d1cffb58-db40-4bc3-be86-976a90edc020.png)

I also tried the branch that is supposed to fix (#19) this and got this (basically the same, I think):
![image](https://user-images.githubusercontent.com/93591857/231217053-5b03f9d7-078f-41c5-8141-0dbb013e6a66.png)

I did a quick fix for myself and will make a PR in case that is useful for you (it is based on the main branch, so probably needs to be redone, but the changes are just 7 lines so it should be easy to replicate).
I am not experienced with Lua, so I don't know how good the solution is, but it seems to work for me so I thought I could share my findings.

**Comment by Hoffs** [comment]
I took at look and as far as I can see, your issue is a bit different from the one I initially encountered. For some reason trying to get metadata for `LoggingBuilderExtensions.cs` does not return anything. The changes that are in #19 also do not expect this, as you already noticed.

```
[DEBUG][2023-04-14 10:34:46] .../vim/lsp/rpc.lua:285	"rpc.send"	{  id = 3,  jsonrpc = "2.0",  method = "o#/v2/gotodefinition",  params = {    column = 25,    fileName = "/.../Program.cs",    line = 53  }}
[DEBUG][2023-04-14 10:34:46] .../vim/lsp/rpc.lua:388	"rpc.receive"	{  id = 3,  jsonrpc = "2.0",  result = {    Definitions = { {        Location = {          FileName = "$metadata$/Project/Railsbank/.../Assembly/Microsoft/Extensions/Logging/Symbol/Microsoft/Extensions/Logging/LoggingBuilderExtensions.cs",          Range = {            End = {              Column = 48,              Line = 45            },            Start = {              Column = 34,              Line = 45            }          }        },        MetadataSource = {          AssemblyName = "Microsoft.Extensions.Logging",          ProjectName = "...",          TypeName = "Microsoft.Extensions.Logging.LoggingBuilderExtensions"        }      } }  }}
[DEBUG][2023-04-14 10:34:46] .../vim/lsp/rpc.lua:285	"rpc.send"	{  id = 4,  jsonrpc = "2.0",  method = "o#/metadata",  params = {    AssemblyName = "Microsoft.Extensions.Logging",    ProjectName = "...",    TypeName = "Microsoft.Extensions.Logging.LoggingBuilderExtensions",    timeout = 5000  }}
[DEBUG][2023-04-14 10:34:46] .../vim/lsp/rpc.lua:388	"rpc.receive"	{  id = 4,  jsonrpc = "2.0",  result = {}}
```

Like omnisharp returns a location with `MetadataSource`, but that source then returns nothing back. 

I will merge your fix for time being, but will have to dig a bit deeper into this.
